amsoell/foteaux
# RESTful API Implementation

Provided RESTful API endpoints for accessing and updating application data. Primarily read-only, with the ability to update a users own account information.

## Test Coverage
- tests/Feature/Api/FeedTest.php 
    - it_returns_feed 
    - it_returns_errors_for_guests 
- tests/Feature/Api/UserFollowTest.php
    - it_follows_users it_unfollows_users
    - it_returns_error_when_following_self
- tests/Feature/Api/UserMediaTest.php 
    - it_returns_user_media 
    - it_returns_other_user_media 
- tests/Feature/Api/UserTest.php 
    - it_returns_user_profile_data 
    - it_returns_other_user_profile_data 
    - it_allows_updates_to_own_profile 
    - it_does_not_allow_updates_to_other_profile 
    - it_does_not_require_all_attributes_when_updating 
- tests/Unit/Actions/FollowTest.php 
    - it_follows_users 
    - it_unfollows_users

## Acceptance Testing Steps
1. Generate your access token at `/user/api-tokens`
1. Using the [provided Paw or Postman files](https://github.com/amsoell/foteaux/files/6912263/foteaux-api-tests.zip), test each of the following API endpoints to ensure their functionality. Be sure to update the request headers to include the `Authorization: Bearer xyz` header.
    - GET /api/v1/feed
    - GET /api/v1/users/{username}
    - PATCH /api/v1/users/{username}
    - GET /api/v1/users/{username}/media
    - POST /api/v1/users/{username}/follow
    - DELETE /api/v1/users/{username}/follow
1. Generate the API documentation with `php artisan l5-swagger:generate`
1. Navigate to `/api/v1` to confirm the documentation has generated correctly
